transparent enum diagnostic should suppress field count when >1 variant

Consider this code:

```rust
#![feature(transparent_enums)]

use std::mem::size_of;

fn main() {
    #[repr(transparent)]
    enum VisibleE8 {
        A(u8), B(u8), C(u8), D(u8), E(u8), F(u8), G(u8), H(u8), I(u8), J(u8),
        K(u8), L(u8), M(u8), N(u8), O(u8), P(u8), Q(u8), R(u8), S(u8), T(u8),
    }

    println!("VisibleE8: {}", size_of::<VisibleE8>());
}
```

When you try to compile this, you get two error diagnostics:

```
error[E0731]: transparent enum needs exactly one variant, but has 20
 --> src/main.rs:7:5
  |
7 |     enum VisibleE8 {
  |     ^^^^^^^^^^^^^^ needs exactly one variant, but has 20
8 |         A(u8), B(u8), C(u8), D(u8), E(u8), F(u8), G(u8), H(u8), I(u8), J(u8),
  |         -----  -----  -----  -----  -----  -----  -----  -----  -----  -----
9 |         K(u8), L(u8), M(u8), N(u8), O(u8), P(u8), Q(u8), R(u8), S(u8), T(u8),
  |         -----  -----  -----  -----  -----  -----  -----  -----  -----  ----- too many variants in `main::VisibleE8`

error[E0690]: the variant of a transparent enum needs exactly one non-zero-sized field, but has 20
 --> src/main.rs:7:5
  |
7 |     enum VisibleE8 {
  |     ^^^^^^^^^^^^^^ needs exactly one non-zero-sized field, but has 20
8 |         A(u8), B(u8), C(u8), D(u8), E(u8), F(u8), G(u8), H(u8), I(u8), J(u8),
  |           --     --     --     --     --     --     --     --     --     -- this field is non-zero-sized
  |           |      |      |      |      |      |      |      |      |
  |           |      |      |      |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      this field is non-zero-sized
  |           |      |      |      this field is non-zero-sized
  |           |      |      this field is non-zero-sized
  |           |      this field is non-zero-sized
  |           this field is non-zero-sized
9 |         K(u8), L(u8), M(u8), N(u8), O(u8), P(u8), Q(u8), R(u8), S(u8), T(u8),
  |           --     --     --     --     --     --     --     --     --     -- this field is non-zero-sized
  |           |      |      |      |      |      |      |      |      |
  |           |      |      |      |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      |      this field is non-zero-sized
  |           |      |      |      |      this field is non-zero-sized
  |           |      |      |      this field is non-zero-sized
  |           |      |      this field is non-zero-sized
  |           |      this field is non-zero-sized
  |           this field is non-zero-sized

error: aborting due to 2 previous errors
```


The first diagnostic is exactly describing the problem.

The second diagnostic is irrelevant, and arguably confusing (at least in my mental model of enums, the field count of one variant should be considered independently of the counts for all of the enum's other variants).

--- a/compiler/rustc_hir_analysis/src/check/check.rs
+++ b/compiler/rustc_hir_analysis/src/check/check.rs
@@ -1060,10 +1060,8 @@ pub(super) fn check_transparent<'tcx>(tcx: TyCtxt<'tcx>, adt: ty::AdtDef<'tcx>) 
 
     if adt.variants().len() != 1 {
         bad_variant_count(tcx, adt, tcx.def_span(adt.did()), adt.did());
-        if adt.variants().is_empty() {
-            // Don't bother checking the fields. No variants (and thus no fields) exist.
-            return;
-        }
+        // Don't bother checking the fields.
+        return;
     }
 
     // For each field, figure out if it's known to be a ZST and align(1), with "known"

--- a/src/test/ui/repr/transparent-enum-too-many-variants.rs
+++ b/src/test/ui/repr/transparent-enum-too-many-variants.rs
@@ -1,0 +1,10 @@
+use std::mem::size_of;
+
+#[repr(transparent)]
+enum Foo { //~ ERROR E0731
+    A(u8), B(u8),
+}
+
+fn main() {
+    println!("Foo: {}", size_of::<Foo>());
+}

--- a/src/test/ui/repr/transparent-enum-too-many-variants.stderr
+++ b/src/test/ui/repr/transparent-enum-too-many-variants.stderr
@@ -1,0 +1,11 @@
+error[E0731]: transparent enum needs exactly one variant, but has 2
+  --> $DIR/transparent-enum-too-many-variants.rs:4:1
+   |
+LL | enum Foo {
+   | ^^^^^^^^ needs exactly one variant, but has 2
+LL |     A(u8), B(u8),
+   |     -      - too many variants in `Foo`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0731`.

